osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -64,44 +65,51 @@ public class ChannelManager : PollingComponent, IChannelPostTarget
         public IBindableList<Channel> AvailableChannels => availableChannels;
 
         private readonly IAPIProvider api;
+        private readonly NotificationsClientConnector connector;
 
         [Resolved]
         private UserLookupCache users { get; set; }
 
-        public readonly BindableBool HighPollRate = new BindableBool();
-
-        private readonly IBindable<bool> isIdle = new BindableBool();
+        private readonly IBindable<APIState> apiState = new Bindable<APIState>();
+        private bool channelsInitialised;
+        private ScheduledDelegate ackDelegate;
 
-        public ChannelManager(IAPIProvider api)
+        public ChannelManager(IAPIProvider api, NotificationsClientConnector connector)
```

Comment:
Kinda weird passing the new parameter here even though it can be fetched directly from `IAPIProvider`. Was there a reason behind this?

Reply:
It's uh... legacy code :smile: . It can be constructed inside this class now.